src/table/table.js

```diff
@@ -241,21 +241,40 @@ class TableExpandedRowContent extends Component {
 
   componentDidUpdate() {
     if (!this.el) return;
-    if (!this.props.open) {
-      this.el.style.setProperty('--mt-table-cellHeight', 0);
-      return;
-    }
 
-    if (!this.props.open) return;
+    const initialBb = this.el.getBoundingClientRect();
     this.el.style.transition = 'none';
     this.el.style.display = 'block';
     this.el.style.height = 'auto';
     const bb = this.el.getBoundingClientRect();
     this.el.style.height = null;
     this.el.style.transition = null;
     this.el.style.display = null;
+
+    if (!this.props.open) {
+      if (initialBb.height) {
+        this.el.style.setProperty(
+          '--mt-table-cellHeight',
+          `${initialBb.height}px`
+        );
+        requestAnimationFrame(() => {
+          this.el.style.setProperty('--mt-table-cellHeight', 0);
```

Comment:
Check for the existence of `this.el` here. It could be unmounted.

Reply:
gooooood catch!